src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/MessageBody.cs

```diff
@@ -273,6 +274,22 @@ private void TryStartTimingReads()
             }
         }
 
+        private void TryPauseTimingReads()
+        {
+            if (!RequestUpgrade)
```

Comment:
Should this only apply after the request has actually been upgraded? This value is true even if the app does not accept the upgrade request.